index.jsx

```diff
@@ -41,20 +41,26 @@ export function FiltersProvider({ children }) {
         return () => {
             window.onhashchange = null;
         };
-    }, [1]);
+    }, []);
 
     function setUserFilters(filtersObject) {
+        const filtersObjectWithExistingVales = Object.keys(filtersObject).filter(element => {
```

Comment:
I think `entries` instead of `keys` can be more suitable in this case, because you'll have access to value without referencing to `filtersObject`. This is also much more functional approach